src/codegen/compile.ml

```diff
@@ -2735,29 +2738,46 @@ module Object = struct
     )
 
   (* Returns a pointer to the object field (possibly following the indirection) *)
-  let idx_hash env indirect =
+  let idx_hash env m indirect =
     if indirect
-    then Func.share_code2 env "obj_idx_ind" (("x", I32Type), ("hash", I32Type)) [I32Type] (fun env get_x get_hash ->
+    then
+      let name = Printf.sprintf "obj_idx_ind<%d>" m in
+      Func.share_code2 env name (("x", I32Type), ("hash", I32Type)) [I32Type] (fun env get_x get_hash ->
       get_x ^^ get_hash ^^
-      idx_hash_raw env ^^
+      idx_hash_raw env m ^^
       load_ptr ^^ compile_add_const (Int32.mul MutBox.field Heap.word_size)
     )
-    else idx_hash_raw env
+    else idx_hash_raw env m
 
   (* Determines whether the field is mutable (and thus needs an indirection) *)
   let is_mut_field env obj_type s =
     let _, fields = Type.as_obj_sub [s] obj_type in
     Type.is_mut (Type.lookup_val_field s fields)
 
+  (* Computes a lower bound for the linear index of a field in an object *)
+  let field_lower_bound env obj_type s =
```

Comment:
I wonder if you can actually assume the fields are sorted and simplify this code a bit.
Might be worth checking if check_ir enforces sortedness. 


(But if in any doubt, just keeping sorting as is)

Reply:
Good point, since I took the sorting lines from `Serialization` when I started with this, it might be redundant here! Will test.

Reply:
They are sorted by label string! Which is different from hash magnitude order... See the failures in 27438402e.

Reply:
Dooh, sorry about that.